### ISSUE TITLE
Empty space not being re-utilized

### ISSUE THREAD

**Issue by mohib-irshad-10p** [objectbox/objectbox-java#856]
As mentioned in the comments of ticket https://github.com/objectbox/objectbox-java/issues/584, I did the following:

- I started counting the no. of records stored and limited it to 200 which max outs at 15MB. All of my records are almost the same size, so I can safely assume 15MB == 200 records.
- I set the size of DB to 30MB.
- On reaching 200 records, I started deleting oldest records but the size kept growing upto the point where it crossed 30MB and crashed with DbFullException.

I've read in different places that Objectbox is supposed to utilize space which it retained even after the data in it was deleted but I've never experienced it myself. Communicated with you guys on this issue as well and provided my workaround: #552 (comment)

Is there a setting which I am missing to enable empty occupied space utilization? I can only delete one record at a time. Please somebody help, its sort of emergency please :(

**Comment by greenrobot** [comment]
Have you verified object counts? Have you checked the contents of your DB with object browser?

All looks like expected? Then, can you please post the page fill stats from object browser for a DB at 15MB with 200 records, and for a DB at 30 MB with 200 records?

**Comment by greenrobot-team** [comment]
FYI https://docs.objectbox.io/data-browser

**Comment by mohib-irshad-10p** [comment]
Thank you v much for responding @greenrobot @greenrobot-team 
Yes, everything looks as expected in the data browser. I did a fresh install of application and started dumping data in it.
In this particular instance, 200 records took ~11MB and the next 200 records increased the size to ~22MB.
Following is the Data and Stats view on first 200 records:
![Data_11_MB](https://user-images.githubusercontent.com/10526654/80988957-b1830080-8e4d-11ea-987f-75e22d29156d.png)
![Stats_11_MB](https://user-images.githubusercontent.com/10526654/80988962-b34cc400-8e4d-11ea-9892-980d2119a81e.png)

Following is Data and Stats view on next 200 records which got inserted by deleting old data:
![Data_22_MB](https://user-images.githubusercontent.com/10526654/80988959-b2b42d80-8e4d-11ea-820d-8a22a10b20a9.png)
![Stats_22_MB](https://user-images.githubusercontent.com/10526654/80988963-b3e55a80-8e4d-11ea-8fdf-19b6c6de12c6.png)

**Comment by mohib-irshad-10p** [comment]
@greenrobot @greenrobot-team 
Thank you very much for responding. While I was gathering data for you via Data Browser, I noticed that when you delete a record from the database, it only deletes that record and not its nested entries, and I was, for some reason, under the impression that child objects would also get deleted.
After manually deleting all the child entries, database size remained consistent.
This was definitely a mistake on my end. Thank you again for your time. Closing this ticket.
